test/staged.jl

```diff
@@ -156,7 +156,7 @@ end
 @generated function _g_f_with_inner(x)
     :(y->y)
 end
-@test (_g_f_with_inner(1))(8) == 8
+@test_throws ErrorException _g_f_with_inner(1)
 
```

Comment:
added by @JeffBezanson in https://github.com/JuliaLang/julia/commit/b43c5e0fba738a188f2b5396ef7a9c51a8dbc3ee - presumably this was needed for something?

I hate to backport new restrictions, even if it's just an earlier error on code that may be invalid. Will have to see how many packages are actually doing this. What's allowed in generated functions is starting to become a pretty restrictive subset of the language, and I don't think people have been aware of that or writing them that way.